### PR TITLE
update SDK folder structure to reflect 4/4/2018 notes

### PR DESCRIPTION
**Pull request by davidkline-ms**
Overview
---
The current MixedRealityToolkit-SDK folder implies a monolithic SDK package.

During the 4/4/2018 shiproom, it was agreed that the SDK folder needs to be easy to componentize, allowing consumers of the MRTK to opt-in to features. 

Changes
---
- Create Features folder and with Input sub-tree
- Move Audio, Fonts, etc to Resources folder
- Update assembly definition to create Microsoft.MixedReality.Toolkit.SDK.Input

### CODE REVIEW

File: Assets/MixedRealityToolkit-SDK/Inspectors/MixedRealityToolkit.SDK.Inspectors.asmdef [hunk 1]
```diff
@@ -3,7 +3,7 @@
     "references": [
         "Microsoft.MixedReality.Toolkit",
         "Microsoft.MixedReality.Toolkit.InputSystem",
-        "Microsoft.MixedReality.Toolkit.SDK"
+        "Microsoft.MixedReality.Toolkit.SDK.Input"
```
Comment (StephenHodgson) [inline Assets/MixedRealityToolkit-SDK/Inspectors/MixedRealityToolkit.SDK.Inspectors.asmdef]: Inspectors are for the SDK level. 

We will need another custom assembly that targets the sdk layer or rename the SDK.Inspectors to only reference SDK.Input.

**Comment by davidkline-ms** [comment]
@StephenHodgson,
The idea is to have an assembly per feature, so as to allow users to opt-in to specific features. Is that an incorrect assumption?

File: Assets/MixedRealityToolkit-SDK/Features/Input/MixedRealityToolkit.SDK.Input.asmdef [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-    "name": "Microsoft.MixedReality.Toolkit.SDK",
+    "name": "Microsoft.MixedReality.Toolkit.SDK.Input",
```
Comment (StephenHodgson) [inline Assets/MixedRealityToolkit-SDK/Features/Input/MixedRealityToolkit.SDK.Input.asmdef]: We probably need both a Microsoft.MixedReality.Toolkit.SDK for the base assembly as well as Microsoft.MixedReality.Toolkit.SDK.Input

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit-SDK/Features/Input/MixedRealityToolkit.SDK.Input.asmdef]: I like this. We should consider putting common scripts in Microsoft.MixedReality.Toolkit.SDK (when we have them in the tree)

**Comment by StephenHodgson** [comment]
> The idea is to have an assembly per feature, so as to allow users to opt-in to specific features. Is that an incorrect assumption?

That is correct.  I was thinking especially so for the internal classes, as for the SDK, we may still need a base SDK assembly.  Even so the SDK.Inspectors assembly is not incorrectly referencing an assembly it shouldn't.  We should either move and rename it for the Input feature specifically or create a base SDK assembly and fix its reference.

File: Assets/MixedRealityToolkit-SDK/Inspectors/MixedRealityToolkit.SDK.Inspectors.asmdef (hunk 1 above)
Comment (davidkline-ms) [reply Assets/MixedRealityToolkit-SDK/Inspectors/MixedRealityToolkit.SDK.Inspectors.asmdef]: Wouldnt inspectors reference all SDK assemblies?

File: Assets/MixedRealityToolkit-SDK/Features/Input/MixedRealityToolkit.SDK.Input.asmdef (hunk 2 above)
Comment (davidkline-ms) [reply Assets/MixedRealityToolkit-SDK/Features/Input/MixedRealityToolkit.SDK.Input.asmdef]: I can start adding in some Common scripts before we merge this and create the base assembly.

Comment (StephenHodgson) [reply Assets/MixedRealityToolkit-SDK/Features/Input/MixedRealityToolkit.SDK.Input.asmdef]: Its no worries, having the custom assembly ahead of time shouldn't impact anything.

Comment (StephenHodgson) [reply Assets/MixedRealityToolkit-SDK/Features/Input/MixedRealityToolkit.SDK.Input.asmdef]: Let's add new scripts in a separate PR

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit-SDK/Features/Input/MixedRealityToolkit.SDK.Input.asmdef]: Having it there displayed an error in Unity which is why I removed it.

Comment (StephenHodgson) [reply Assets/MixedRealityToolkit-SDK/Features/Input/MixedRealityToolkit.SDK.Input.asmdef]: Also, I'm not so sure about starting on the SDK layer until we've finished most of the work on the Core feature areas.

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit-SDK/Features/Input/MixedRealityToolkit.SDK.Input.asmdef]: This is actually intended for beta. Don't have a branch for that work... yet :)

**Comment by davidkline-ms** [comment]
@StephenHodgson, in shiproom, it sounded like folks were ok taking this... did I hear correctly?

**Comment by SimonDarksideJ** [comment]
Correct, we agreed it was safe to merge this in at this early stage.

**Comment by SimonDarksideJ** [comment]
That should confirm it :D

**Comment by StephenHodgson** [comment]
Yo, Why did this get merged when there was CRs?

**Comment by StephenHodgson** [comment]
@davidkline-ms I had an open PR on your branch for CRs

**Comment by davidkline-ms** [comment]
I didn't merge this... :\

**Comment by davidkline-ms** [comment]
Let me look at my branch and I can submit another PR

**Comment by StephenHodgson** [comment]
Just opened https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/2213

**Pull request merged**